evalml/tests/automl_tests/test_iterative_algorithm.py

```diff
@@ -98,19 +97,20 @@ def test_iterative_algorithm_init(
     assert algo.batch_number == 0
     assert algo.default_max_batches == 1
     estimators = get_estimators(problem_type)
-    decomposer = [STLDecomposer] if is_regression(problem_type) else []
+    decomposer = [True, False] if is_regression(problem_type) else [True]
```

Comment:
Can you add a comment clarifying why you're using true false instead of the decomposer name?

Reply:
I think it would be better to just parametrize the `include_decomposer` argument here - this and the below section are confusing to read out of context

Reply:
For this test, we are basically only checking the number of pipelines matches up. Before, we only needed to add the decomposer one once since there was one estimator type (VARMAX).

Now, since we have multiple estimator types, each estimator type will have one pipeline with a decomposer and another without a decomposer. As such, we need to have this [True, False] and to iterate through it in order to generate the correct number of pipelines.

Reply:
I think a clarifying comment would be useful here 👍

Reply:
Added a short one to this test